src/commands/retrieve/metadata.ts

```diff
@@ -138,8 +148,38 @@ export default class RetrieveMetadata extends SfCommand<RetrieveMetadataResult>
     const fileResponses = result?.getFileResponses() || [];
 
     if (!flags.json) {
-      displaySuccesses(result);
+      if (result.response.status === 'Succeeded') {
+        await this.displayResults(result, flags);
+      } else {
+        throw new SfdxError(
+          getString(result.response, 'errorMessage', result.response.status),
+          getString(result.response, 'errorStatusCode', 'unknown')
+        );
+      }
     }
     return fileResponses;
   }
+  private async displayResults(result: RetrieveResult, flags): Promise<void> {
+    if (!getBoolean(flags, 'json', false)) {
+      if (result.response.status === 'Succeeded') {
+        displaySuccesses(result);
+        displayPackages(result, await this.getPackages(result, flags));
+      } else {
+        throw new SfdxError(
+          getString(result.response, 'errorMessage', result.response.status),
+          getString(result.response, 'errorStatusCode', 'unknown')
+        );
```

Comment:
This isn't necessary since the same check and error throw is already done before `displayResults()` is called.

Reply:
Sorry, I relocated the throw and forgot to refactor original. Thanks for catching.